flexget/utils/imdb.py

```diff
@@ -37,22 +37,26 @@ def is_imdb_url(url):
     return re.match(r'https?://[^/]*imdb\.com/', url)
 
 
-def is_valid_imdb_title_id(value=None):
+def is_valid_imdb_title_id(value):
     """
     Return True if `value` is a valid IMDB ID for titles (movies, series, etc).
     """
-    if not isinstance(value, basestring):
+    if value is None:
         return False
+    if not isinstance(value, basestring):
+        raise TypeError("is_valid_imdb_title_id expects a string but got {0}".format(type(value)))
     # IMDB IDs for titles have 'tt' followed by 7 digits
     return re.match('tt[\d]{7}', value) is not None
 
 
-def is_valid_imdb_person_id(value=None):
+def is_valid_imdb_person_id(value):
     """
     Return True if `value` is a valid IMDB ID for a person.
     """
-    if not isinstance(value, basestring):
+    if value is None:
```

Comment:
You are fast!
I was writing the message explaining the reason and after I finished I saw a new comment.

The reason is that we can write something as

```
if is_valid_imdb_title_id(entry.get('nfo_id')):
    do_some_work_using_nfo_id_field()
```

In case the entry does not have the field then `entry.get` returns `None`,
`is_valid_imdb_title_id` returns `False` and we don't try to use the field
value. If `is_valid_imdb_title_id` raises an exception for `None` then everyone
would have to test for that.